buildingspy/development/validator.py

```diff
@@ -177,3 +179,377 @@ def _validateHTML(self, moFile):
+        """ 
+        Validate parameter settings.
+    
+        :param name: Parameter name.
+        :param mos_files: List of mos files.
+        :param root_dir: Root directory.
+    
+         """
+
+        N_mos_defect=0
+    
+        j = 1
+        for mos_file in mos_files:
+            j += 1
+            
+            f = open(mos_file,"r", encoding="utf8")
+            
+            content = f.readlines()
+            found = False
+            i = 0
+            while found == False and i<len(content):
+                l = content[i]
+                if "simulateModel(" in l.replace(" ", ""):
+                    line = l
+                    found = True
+                i += 1
+            
+            try:
+                if ""+name+"="+name+"" in line.replace(" ", ""):
+                    value = ""+name+""
+                    self._wrong_literal(mos_file ,name)
+                   
+                if ""+name+"="+"" in line.replace(" ", ""):
+                    pTime    = re.compile(r"[\d\S\s.,]*("+name+"=)([\d]*[.]*[\d]*[eE]*[+|-]*[\d]*[*]*[\d]*[.]*[\d]*[eE]*[+|-]*[\d]*)")
+                    mTime    = pTime.match(line)
+                    value = mTime.group(2)
+                    self._check_tolerance(content, name, value, mos_file)     
+                else:
+                    found = False
+                    while found == False and i<len(content):
+                        line = content[i]
+                        i += 1
+                        # Remove white spaces
+                        line.replace(" ", "")
+                        
+                        if ""+name+"="+"" in line.replace(" ", ""):
+                            found = True
+                            pTime    = re.compile(r"[\d\S\s.,]*("+name+"=)([\d]*[.]*[\d]*[eE]*[+|-]*[\d]*[*]*[\d]*[.]*[\d]*[eE]*[+|-]*[\d]*)[\S\s.,]*")
+                            mTime    = pTime.match(line)
+                            value = mTime.group(2)
+                            self._check_tolerance(content, name, value, mos_file)  
+                            #startTime = startTime[:-1]
+                        if ""+name+"="+name+"" in line.replace(" ", ""):
+                            value = ""+name+""
+                            self._wrong_literal(mos_file, name)
+                    if found == False:
+                        if (name=="startTime"):
+                            value = "0.0"
+                        elif (name=="stopTime"):
+                            value="1.0"
+                        elif(name=="tolerance"):
+                            value="0.0"
+                            print("Should be coming here")
+                            self._wrong_parameter (mos_file, name, value)
+    
+            except AttributeError:
+                value = "NA"
+                N_mos_defect += 1
+                
+            if value != "NA" and value != ""+name+"":   
```

Comment:
Why `""+name"+""`, and why NA? Is this a string variable that is used to test something? If so, please use `None` or a boolean flag.